lib/util/timestamp.js

```diff
@@ -1,16 +1,84 @@
-// Get a date object in the correct format, without requiring a full out library
-// like "moment.js".
-function yyyymmddhhmmss() {
-  const d = new Date();
-
-  return (
-    d.getFullYear().toString() +
-    (d.getMonth() + 1).toString().padStart(2, '0') +
-    d.getDate().toString().padStart(2, '0') +
-    d.getHours().toString().padStart(2, '0') +
-    d.getMinutes().toString().padStart(2, '0') +
-    d.getSeconds().toString().padStart(2, '0')
-  );
+function yyyymmddhhmmss(...time) {
+	const rYear = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+	const timeZones = {
+		'60':   'ECT', '120':  'EET', '180':  'EAT', '210':  'MET',
+		'240':  'NET', '300':  'PLT', '330':  'IST', '360':  'BST',
+		'420':  'VST', '480':  'CTT', '540':  'JST', '570':  'ACT',
+		'600':  'AET', '660':  'SST', '720':  'NST', '-660': 'MIT',
+		'-600': 'HST', '-540': 'AST', '-480': 'PST', '-420': 'MST',
+		'-360': 'CST', '-300': 'EST', '-240': 'PRT', '-210': 'CNT',
+		'-180': 'BET', '-60':  'CAT'
+	}
+	
+	let d = new Date()
+	let offset = d.getTimezoneOffset()
+	let zoneName = 'UTC'
+	if (time.length > 0) {
+		//'2016-02-28T21:12:45.678'
+		let regex = new RegExp('^[0-9]{4}-[0-9]{2}-[0-9]{2}T([0-9]{2}:){2}[0-9]{2}\.[0-9]{3}$')
+		if ()
```

Comment:
is this a leftover code?

Reply:
This is something I was working on very late as I was working with one of my coworkers to validate times given in, as this method allows for no parameters or two parameters